lib/services/summary.js

```diff
@@ -37,26 +39,30 @@ class SummaryService {
     };
 
     if (this._fullSummary(req) && !decoder.canSummarizeQuickly()) {
+      req.log.info("Making full summary");
+
       req
-      .pipe(decoder)
-      .on('error', onErr)
-      .pipe(new Merger(disk, [], true))
-      .on('error', onErr)
-      .on('end', (layers) => {
-        ok(layers.map((layer) => layer.toJSON()));
-      });
-    } else {
-      decoder.on('data', (_datum) => {
-        decoder.pause();
-        decoder.summarize((err, summary) => {
-          if (err) return res.status(400).send(JSON.stringify(err));
-          return ok(summary);
+        .pipe(decoder)
+        .on('error', onErr)
+        .pipe(new Merger(disk, [], true))
+        .on('error', onErr)
+        .on('end', (layers) => {
+          ok(layers.map((layer) => layer.toJSON()));
         });
-      });
+    } else {
+      req.log.info("Making abbreviated summary");
 
       req
-      .pipe(decoder)
-      .pipe(new DevNull());
+        .pipe(decoder)
+        .on('data', (_datum) => {
+          decoder.pause();
+          decoder.summarize((err, summary) => {
+            if (err) return res.status(400).send(JSON.stringify(err));
+            return ok(summary);
+          });
+        })
+        .on('error', onErr)
+        .pipe(new DevNull());
```

Comment:
In node, a stream doesn't start flowing until someone tries to read from it. So we need to attach a sink (in this case DevNull, which just throws everything away)  to get the stream to start. 

Some relevant code in the shapefile decoder is here:

```javascript
  _read() {
    if (!this._readableState.emittedReadable && !this._isPushing) {
      this.once('readable', this._startPushing.bind(this));
    } else if (!this._isPushing) {
      this._startPushing();
    }
  }
```

`_read` needs to be implemented in readable all streams, and will be called (eventually) when `this` stream's `pipe` method is called. Then, when data can be read (ie the stream is `readable`) the stream starts pushing data into the queue, which emits `data` events. 

This summary thing is a little confusing because we can only do a summary once the blob is buffered to disk (and then we read the shapefile headers), so that's why it needs to wait for a `data` event to be emitted before making a summary.